File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/repositories/UserProblemStatusRepository.java
package io.github.sanyavertolet.edukate.backend.repositories;

import io.github.sanyavertolet.edukate.backend.entities.UserProblemStatus;
import org.springframework.stereotype.Repository;
import reactor.core.publisher.Mono;

@Repository
public interface UserProblemStatusRepository extends ReactiveReadOnlyRepository<UserProblemStatus, String> {
    Mono<UserProblemStatus> findByUserIdAndProblemId(String userId, String problemId);
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/repositories/UserProblemStatusRepository.kt
package io.github.sanyavertolet.edukate.backend.repositories

import io.github.sanyavertolet.edukate.backend.entities.UserProblemStatus
import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono

@Repository
interface UserProblemStatusRepository : ReactiveReadOnlyRepository<UserProblemStatus, String> {
    fun findByUserIdAndProblemId(userId: String, problemId: String): Mono<UserProblemStatus>
}
